cmd/srcd/cmd/root.go

```diff
@@ -80,3 +85,35 @@ func initConfig() {
 		fmt.Println("Using config file:", viper.ConfigFileUsed())
 	}
 }
+
+var logMsgRegex, _ = regexp.Compile(`.*msg="(.+)"`)
+
+func logAfterTimeout(header string) chan struct{} {
+	logs, err := daemon.GetLogs()
+	if err != nil {
+		logrus.Fatalf("could not get logs from server container: %v", err)
+	}
+
+	started := make(chan struct{})
+	go func() {
+		select {
+		case <-time.After(3 * time.Second):
+			logrus.Info(header)
+			scanner := bufio.NewScanner(logs)
+			for scanner.Scan() {
+				match := logMsgRegex.FindStringSubmatch(scanner.Text())
+				if len(match) == 2 {
+					logrus.Info(match[1])
+				}
+			}
+			if err := scanner.Err(); err != nil && err != context.Canceled {
+				logrus.Fatal(err)
+			}
+			logs.Close()
```

Comment:
seems like you could move the` logs.Close` into a `defer logs.close` at the beginning of the goroutine